sdk/core/Azure.Core/src/Shared/EventSourceEventFormatting.cs

```diff
@@ -18,12 +19,24 @@ public static string Format(EventWrittenEventArgs eventData)
 
             if (eventData.Message != null)
             {
-                return string.Format(CultureInfo.InvariantCulture, eventData.Message, payloadArray);
+                try
+                {
+                    return string.Format(CultureInfo.InvariantCulture, eventData.Message, payloadArray);
+                }
+                catch (FormatException)
+                {
+                }
             }
 
             var stringBuilder = new StringBuilder();
             stringBuilder.Append(eventData.EventName);
 
+            if (!string.IsNullOrWhiteSpace(eventData.Message))
```

Comment:
Why are we appending the eventData.Message here?

Reply:
If theres a case when you do have a message and you got to this part it means you failed formatting and caught the exception.

In this case you still want to print the message and the payload. Just bot in a formatted way.